mis_builder/models/mis_builder.py

```diff
@@ -439,20 +450,36 @@ def _compute(self, lang_id, aep,
                        additional_move_line_filter)
 
         compute_queue = self.kpi_ids
+        recompute_inherit = {}
         recompute_queue = []
         while True:
             for kpi in compute_queue:
                 try:
                     kpi_val_comment = kpi.name + " = " + kpi.expression
                     kpi_eval_expression = aep.replace_expr(kpi.expression)
-                    kpi_val = safe_eval(kpi_eval_expression, localdict)
+
+                    if recomputed_dict:
+                        kpi_val = recomputed_dict.get(kpi_eval_expression)
+                        recomputed_dict = {}
+                    else:
+                        kpi_val = safe_eval(kpi_eval_expression, localdict)
+
                     localdict[kpi.name] = kpi_val
                 except ZeroDivisionError:
                     kpi_val = None
                     kpi_val_rendered = '#DIV/0'
                     kpi_val_comment += '\n\n%s' % (traceback.format_exc(),)
                 except (NameError, ValueError):
-                    recompute_queue.append(kpi)
+                    if '.' in kpi.expression:
+                        report_descript, kpi_name = kpi.expression.split('.')
```

Comment:
@HugoUchoasBorges this `split` is not going to work with complex expressions. There maybe many `.` in the expression.

The design I had in mind when I thought about that last year was something like this:
* add on `mis.report` a m2m with a list of dependent `mis.report`
* the code you add on `mis.report` looks like a great idea
* then at the beginning of _compute (before the recompute loop), evaluate the dependent reports and add their localdict as some sort of dataclass in the localdict of the main report with the dependent report code as key name

In v9+ there is an `evaluate` method on `mis.report` that simplifies the evaluation of dependent report a little bit. You can probably get inspiration there.

Anyway, this PR is cool!